### ISSUE TITLE
DateTime to BSON introduces space instead of 0 (in millisecond)

### ISSUE THREAD

**Issue by shivchawla** [felipenoris/Mongoc.jl#17]
Hello

When assigning `DateTime("2019-02-26T21:08:27.057")` to a BSON document, 0 in millisecond component is replaced by space

```
doc = Mongoc.BSON()
doc["date"] = DateTime("2019-02-26T21:08:27.057")
```
gives

`BSON("{ "date" : { "$date" : "2019-02-26T21:08:27. 57Z" } }")`

Is this desired?

Thanks!

**Comment by felipenoris** [comment]
Hi! In short, this is out of my control, and I guess there's nothing to worry about.

Longer answer... When you print a document to screen, the package will call [bson_as_relaxed_extended_json](http://mongoc.org/libbson/current/bson_as_relaxed_extended_json.html) from libmongoc library.

```julia
julia> doc
BSON("{ "date" : { "$date" : "2019-02-26T21:08:27. 57Z" } }")
```

So, it's just a string output, and there is no data loss. You can recover the same value from the document later, as in:

```julia
julia> doc["date"]
2019-02-26T21:08:27.057
```

You can also print using other method [bson_as_canonical_extended_json](http://mongoc.org/libbson/current/bson_as_canonical_extended_json.html):

```julia
julia> Mongoc.as_json(doc, canonical=true)
"{ \"date\" : { \"\$date\" : { \"\$numberLong\" : \"1551215307057\" } } }"
```

**Comment by felipenoris** [comment]
Looks like this has been fixed in libmongoc 1.14 (https://github.com/mongodb/mongo-c-driver/releases/tag/1.14.0).

**Comment by felipenoris** [comment]
Fixed by #33.
